lib/coinbase/webhook.rb

```diff
@@ -0,0 +1,181 @@
+      # Creates a new webhook for a specified network.
+      #
+      # @param network_id [String] The network ID for which the webhook is created.
+      # @param notification_uri [String] The URI where notifications should be sent.
+      # @param event_type [String] The type of event for the webhook. Must be one of the following:
+      #   - `Coinbase::Webhook::ERC20_TRANSFER_EVENT`
+      #   - `Coinbase::Webhook::ERC721_TRANSFER_EVENT`
+      # @param event_filters [Array<Hash>] Filters applied to the events that determine
+      #   which specific events trigger the webhook. Each filter should be a hash that
+      #   can include keys like `contract_address`, `from_address`, or `to_address`.
+      # @return [Coinbase::Webhook] A new instance of Webhook.
+      #
+      # @example Create a new webhook
+      #   webhook = Coinbase::Webhook.create(
+      #     network_id: :ethereum_mainnet,
+      #     notification_uri: 'https://example.com/callback',
+      #     event_type: 'transaction',
+      #     event_filters: [{ 'contract_address' => '0x...', 'from_address' => '0x...', 'to_address' => '0x...' }]
+      #   )
+      def create(network_id:, notification_uri:, event_type:, event_filters:)
+        model = Coinbase.call_api do
+          webhooks_api.create_webhook(
+            create_webhook_request: {
+              network_id: Coinbase.normalize_network(network_id),
+              notification_uri: notification_uri,
+              event_type: event_type,
+              event_filters: event_filters
+            }
+          )
+        end
+
+        new(model)
+      end
+
+      # Enumerates the webhooks.
+      # The result is an enumerator that lazily fetches from the server, and can be iterated over,
+      # converted to an array, etc...
+      # @return [Enumerable<Coinbase::Webhook>] Enumerator that returns webhooks
+      def list
+        Coinbase::Pagination.enumerate(method(:fetch_webhooks_page).to_proc) do |webhook|
+          Coinbase::Webhook.new(webhook)
+        end
+      end
+
+      private
+
+      def webhooks_api
+        Coinbase::Client::WebhooksApi.new(Coinbase.configuration.api_client)
+      end
+
+      def fetch_webhooks_page(page)
+        webhooks_api.list_webhooks({ limit: DEFAULT_PAGE_LIMIT, page: page })
+      end
+    end
+
+    # Initializes a new Webhook object.
+    #
+    # @param model [Coinbase::Client::Webhook] The underlying Webhook object.
+    # @raise [ArgumentError] If the model is not a Coinbase::Client::Webhook.
+    def initialize(model)
+      raise ArgumentError, 'model must be a Webhook' unless model.is_a?(Coinbase::Client::Webhook)
+
+      @model = model
+    end
+
+    # Returns the ID of the webhook.
+    #
+    # @return [String] The ID of the webhook.
+    def id
+      @model.id
+    end
+
+    # Returns the network ID associated with the webhook.
+    #
+    # @return [String] The network ID of the webhook.
+    def network_id
+      @model.network_id
```

Comment:
We are currently converting this to symbols everywhere, e.g. 

```
      Coinbase.to_sym(@model.network_id)
```

I'm doing a once over to have this use full on Network objects in the 0.1.0 release however so this isn't a blocker for landing, just a blocker for release

Reply:
sg, addressed per suggestion